src/snowflake/cli/plugins/nativeapp/codegen/artifact_processor.py

```diff
@@ -36,18 +35,12 @@ def __init__(self, processor_name: str):
 
 
 class ArtifactProcessor(ABC):
-    @abstractmethod
     def __init__(
         self,
-        project_definition: NativeApp,
-        project_root: Path,
-        deploy_root: Path,
-        generated_root: Path,
+        project: NativeAppProjectModel,
         **kwargs,
     ) -> None:
-        assert project_root.is_absolute()
-        assert deploy_root.is_absolute()
-        assert generated_root.is_absolute()
```

Comment:
I saw that the asserts have been moved from here to the SnowparkAnnotationProcessor class. Do we not want to have this assert at the base class level?

Reply:
So far the Snowpark processor relies heavily on that invariant, but I didn't think it was a property of _all_ processors. In any case, I made it a strong invariant now, so the assertions are really redundant.